tasks/scheduler.go

```diff
@@ -193,28 +197,24 @@ func (s *scheduler) validateAll(tasks []*deliverTxTask) ([]*deliverTxTask, error
 
 		// validated tasks can become unvalidated if an earlier re-run task now conflicts
 		case statusExecuted, statusValidated:
-			conflicts := s.findConflicts(tasks[i])
-
-			if len(conflicts) > 0 {
-				// apply the abort to the multiversion stores
+			if valid, conflicts := s.findConflicts(tasks[i]); !valid {
 				s.invalidateTask(tasks[i])
 
 				// if the conflicts are now validated, then rerun this task
 				if indexesValidated(tasks, conflicts) {
 					res = append(res, tasks[i])
 				} else {
 					// otherwise, wait for completion
-					tasks[i].Conflicts = conflicts
+					tasks[i].Dependencies = conflicts
 					tasks[i].Status = statusWaiting
 				}
-			} else {
-				// no conflicts means this task is validated
+			} else if len(conflicts) == 0 {
 				tasks[i].Status = statusValidated
 			}
```

Comment:
in the case where we are valid but have conflicts, does the task states stay as `executed` then? (just want to make sure my understanding is correct

Reply:
that's right, it remains as executed and not returned for re-execution, and therefore is eligible for re-validation